bin/guard

```diff
@@ -2,31 +2,114 @@
 
 require "pathname"
 
-default = Pathname(ENV.fetch("BUNDLE_GEMFILE", "Gemfile"))
-relative_to_binstub = Pathname(__FILE__).realpath + "../../Gemfile"
-gemfile = [default, relative_to_binstub].detect(&:exist?)
-
-if gemfile && !ENV["RUBYGEMS_GEMDEPS"]
-  ENV["BUNDLE_GEMFILE"] = gemfile.to_s
-  require "rubygems"
-  require "bundler/setup"
-else
-  require "rubygems"
-end
+class GuardReloader
+  class Config
+    def using_rubygems?
+      ENV["RUBYGEMS_GEMDEPS"]
```

Comment:
With the way rubygems is autoloaded in Ruby > 2, you'll get a different set of rubygems files required depending on whether or not you use `RUBYGEMS_GEMDEPS`. Generally, requiring rubygems here is superfluous (in Ruby > 2 at least), except for the fact that I can't prove that `Gem.use_gemdeps` was called *after* setting the `RUBYGEMS_GEMDEPS`. At the very least, this allows running Guard with `RUBYGEMS_GEMDEPS` set from a main script running without rubygems in Ruby itself (with --disable-gems).

Example:

`RUBYGEMS_GEMDEPS=- ruby --disable-gems -e 'require "rubygems"; require "rspec"'`

(Which works. Strangely, it works even if RSpec isn't in the Gemfile...)